src/software/ai/passing/evaluation_test.cpp

```diff
@@ -178,7 +178,8 @@ TEST_F(PassingEvaluationTest, ratePass_enemy_directly_on_pass_trajectory)
     });
     world.updateEnemyTeamState(enemy_team);
```

Comment:
I wrote a test for passing without one-time shot optimization. Not sure why it isn't in here, but here it is:
```
TEST_F(PassingEvaluationTest, ratePass_attempting_to_pass_and_receive_no_shot)
{
    // Test that a pass which does NOT result in a good shot on goal is rated
    // highly if we are rating it as a pass which is intended to be received

    World world = ::Test::TestUtil::createBlankTestingWorld();
    world.mutableFriendlyTeam().updateRobots({
                                       Robot(0, {1, 0}, {0, 0}, Angle::half(), AngularVelocity::zero(),
                                             Timestamp::fromSeconds(0)),
                               });
    world.mutableEnemyTeam().updateRobots({
                                       Robot(0, world.field().enemyGoal(), {0, 0}, Angle::zero(), AngularVelocity::zero(),
                                             Timestamp::fromSeconds(0)),
                                       Robot(1, world.field().enemyGoal() - Vector(0, 0.2), {0, 0}, Angle::zero(), AngularVelocity::zero(),
                                             Timestamp::fromSeconds(0)),
                                       Robot(2, world.field().enemyGoal() + Vector(0, 0.2), {0, 0}, Angle::zero(), AngularVelocity::zero(),
                                             Timestamp::fromSeconds(0)),
                               });

    // Since we're passing from the origin to a point directly in front of the goal,
    // the receiving robot would have to turn 180 degrees to take a shot after
    // receiving the ball
    Pass pass({0, 0}, {1, 0}, avg_desired_pass_speed, Timestamp::fromSeconds(1.0));

    double pass_rating = ratePass(world, pass, std::nullopt, std::nullopt, PassType::RECEIVE_AND_DRIBBLE);
    EXPECT_GE(pass_rating, 0.9);
    EXPECT_LE(pass_rating, 1.0);
}
```

might want to double check things got copied over correctly.

Reply:
Hm, that wasn't in master and it wasn't in the robocup_2019 diff

Reply:
Test has been added